### DISCUSSION TITLE
[@types/plotly.js] Incorrect type for Template data

### DISCUSSION THREAD

**Discussion by dforegger** [DefinitelyTyped/DefinitelyTyped#61459] [Issues with a @types package] [answered]
According to https://plotly.com/javascript/reference/layout/#layout-template the `template.data` property format is  ```data: {[type]: [traceTemplate, ...]}```. That is to say, it's a map with keys of `PlotType` and values of _arrays_ of `PlotData`.

In the plotly.js types here, it is defined as a single (Partial) PlotData:
```
export interface Template {
    data?: { [type in PlotType]?: Partial<PlotData> } | undefined;
    layout?: Partial<Layout> | undefined;
}
```

I'm fairly new to Typescript, but I believe this is supposed to be:
```
data?: { [type in PlotType]?: Partial<PlotData>[] } | undefined;
```

Let me know if I can help.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
you mean like here?
https://github.com/plotly/plotly.js/blob/623fcd1fea9d9bfb86e5e0d44d8047cd8636881c/src/plot_api/template_api.js#L60
which creates (and uses) array objecct.

**Reply by dforegger** [reply]
Yup. As you saw, I threw together a PR for this and it's now resolved. Thanks for the quick review and resolution :)

**Comment by peterblazejewicz** [comment, marked as answer]
fixed via #61462
